packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -144,7 +144,7 @@ const getProps = ({
     fontSize: buttonTypographyTokens.fonts.size[size],
     lineHeight: buttonTypographyTokens.lineHeights[size],
     minHeight: makeSize(buttonMinHeight[size]),
-    iconPadding: hasIcon ? `spacing.${textPadding[size]}` : undefined,
+    iconPadding: hasIcon && children?.trim() ? `spacing.${textPadding[size]}` : undefined,
```

Comment:
We should be adding padding only when there is icon _and_ text. Without this, Icon Only buttons had an extra padding